server/mscalendar/auto_respond_test.go

```diff
@@ -0,0 +1,167 @@
+			autoRespondSetting:         false,
+			autoRespondMessageSetting:  "Hello, I'm in a meeting and will respond to your message as soon as I'm free.",
+		},{
+			name:                       "Recipient user status is set to online",
+			expectedError:              "",
+			recipientActiveEvents:      []string{"active_event_hash"},
+			recipientStatusString:      model.STATUS_ONLINE,
+			autoRespondSetting:         true,
+			autoRespondMessageSetting:  "Hello, I'm in a meeting and will respond to your message as soon as I'm free.",
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			mockStore := mock_store.NewMockStore(ctrl)
+			mockPoster := mock_bot.NewMockPoster(ctrl)
+			mockRemote := mock_remote.NewMockRemote(ctrl)
+			mockPluginAPI := mock_plugin_api.NewMockPluginAPI(ctrl)
+
+			conf := &config.Config{PluginVersion: "x.x.x"}
+			env := Env{
+				Config: conf,
+				Dependencies: &Dependencies{
+					Store:     mockStore,
+					Logger:    &bot.NilLogger{},
+					Poster:    mockPoster,
+					Remote:    mockRemote,
+					PluginAPI: mockPluginAPI,
+				},
+			}
+
+			post := newTestMattermostPost()
+			channel := newTestMattermostChannel()
+			mattermostUserSender := newTestMattermostUser("mattermost_user_sender_id")
+			mattermostUserRecipient := newTestMattermostUser("mattermost_user_recipient_id")
+
+			recipientStatus := newTestMattermostStatus(tc.recipientStatusString)
+			storedRecipient := newTestStoreUser(tc.recipientActiveEvents, "mattermost_user_recipient_id")
+
+			mockPluginAPI.EXPECT().GetMattermostChannel("mattermost_post_channel_id").Return(channel, nil)
+			mockPluginAPI.EXPECT().GetMattermostUsersInChannel("mattermost_post_channel_id", model.CHANNEL_SORT_BY_USERNAME, 0, 2).Return([]*model.User{mattermostUserSender, mattermostUserRecipient}, nil)
+
+			mockStore.EXPECT().LoadUser("mattermost_user_sender_id").Return(nil, errors.New("user not found"))
+			mockStore.EXPECT().LoadUser("mattermost_user_recipient_id").Return(storedRecipient, nil)
+
+			mockPluginAPI.EXPECT().GetMattermostUserStatus("mattermost_user_recipient_id").Return(recipientStatus, nil)
+
+			if (tc.recipientStatusString != model.STATUS_ONLINE) {
+				mockStore.EXPECT().GetSetting("mattermost_user_recipient_id", store.AutoRespondSettingID).Return(tc.autoRespondSetting, nil)
+
+				if (tc.autoRespondSetting.(bool) && len(tc.recipientActiveEvents) > 0) {
+					mockStore.EXPECT().GetSetting("mattermost_user_recipient_id", store.AutoRespondMessageSettingID).Return(tc.autoRespondMessageSetting, nil)
+
+					if (tc.autoRespondMessageSetting != "") {
+						mockPoster.EXPECT().Ephemeral("mattermost_user_sender_id", "mattermost_post_channel_id", "Hello, I'm in a meeting and will respond to your message as soon as I'm free.")
+					} else {
+						mockPoster.EXPECT().Ephemeral("mattermost_user_sender_id", "mattermost_post_channel_id", "This user is currently in a meeting.")
```

Comment:
Instead of having this conditional logic in the tests (as we should not be duplicating the actual code's logic in the test), can we have a `expectedDMMessage` in the test case data? If the value is `""` then we can assume that the DM was not supposed to be sent.